deps/eventlog/src/ccel/device_paths/messaging_parser.rs

```diff
@@ -0,0 +1,207 @@
+use super::{DevicePathParser, DeviceSubTypeParser};
+use anyhow::{anyhow, bail, Result};
+use scroll::{Pread, LE};
+use std::net::Ipv4Addr;
+
+use std::collections::HashMap;
+use std::sync::LazyLock;
+
+/// Parser for Type 3 - Messaging Device Path.
+/// Define in section 10.3.4 of <https://uefi.org/specs/UEFI/2.10/10_Protocols_Device_Path_Protocol.html#messaging-device-path>
+static MESSAGING_SUBTYPE_HANDLERS: LazyLock<
+    HashMap<u8, Box<dyn DeviceSubTypeParser + Send + Sync>>,
+> = LazyLock::new(|| {
+    let mut handlers: HashMap<u8, Box<dyn DeviceSubTypeParser + Send + Sync>> = HashMap::new();
+    handlers.insert(0x15, Box::new(FibreChannelExParser));
+    handlers.insert(0x05, Box::new(UsbParser));
+    handlers.insert(0x0B, Box::new(MacParser));
+    handlers.insert(0x0C, Box::new(Ipv4Parser));
+    // Add more subtype parsers here as needed
+    handlers
+});
+
+pub struct MessagingParser;
+
+impl DevicePathParser for MessagingParser {
+    fn parse(&self, sub_type: u8, data: &[u8]) -> Result<String> {
+        MESSAGING_SUBTYPE_HANDLERS
+            .get(&sub_type)
+            .ok_or_else(|| anyhow!("Unknown Messaging subtype: {:#04x}", sub_type))?
+            .parse(data)
+    }
+}
+
+pub struct FibreChannelExParser;
+
+impl DeviceSubTypeParser for FibreChannelExParser {
+    fn parse(&self, data: &[u8]) -> Result<String> {
+        if data.len() < 16 {
+            bail!("Fibre channel data is too short");
+        }
+
+        let wwn = &data[0..8];
+        let lun = &data[8..16];
+
+        Ok(format!(
+            "FibreEx(0x{},0x{})",
+            hex::encode(wwn),
+            hex::encode(lun)
+        ))
+    }
+}
+
+pub struct UsbParser;
+
+impl DeviceSubTypeParser for UsbParser {
+    fn parse(&self, data: &[u8]) -> Result<String> {
+        if data.len() < 2 {
+            bail!("USB data is too short");
+        }
+
+        let parent_hub_port_num = &data[0];
+        let controller_int_number = &data[1];
+
+        Ok(format!(
+            "USB({},{})",
+            parent_hub_port_num, controller_int_number
+        ))
+    }
+}
+
+pub struct MacParser;
+
+impl DeviceSubTypeParser for MacParser {
+    fn parse(&self, data: &[u8]) -> Result<String> {
+        if data.len() < 32 {
```

Comment:
shouldn't it be [exactly](https://uefi.org/specs/UEFI/2.10/10_Protocols_Device_Path_Protocol.html#mac-address-device-path) 32 bytes?

Reply:
Indeed -> fixed